src/js/media-theme-element.js

```diff
@@ -27,11 +29,30 @@ export class MediaThemeElement extends window.HTMLElement {
     super();
     this.renderRoot = this.attachShadow({ mode: 'open' });
 
-    const observer = new MutationObserver(() => this.render());
+    const observer = new MutationObserver((mutationList) => {
+      if (mutationList.some((mutation) => {
+        if (mutation.type !== 'attributes') return false;
+
+        // Render on each attribute change of the `<media-theme>` element.
+        if (this.hasAttribute(mutation.attributeName)) return true;
+
+        // Render if this attribute is directly observed.
+        if (observedMediaAttributes[mutation.attributeName]) return true;
```

Comment:
question(blocking): Should we filter down to only monitor based on specific [`target`s](https://developer.mozilla.org/en-US/docs/Web/API/MutationRecord/target)?

Reply:
good point! I'll add a check for `media-controller` only then.